driver-core/src/test/java/com/datastax/driver/core/ClusterInitTest.java

```diff
@@ -278,6 +278,69 @@ public void should_not_abort_init_if_host_does_not_support_protocol_version() {
       scassandraCluster.stop();
     }
   }
+  /**
+   * Ensures that if an error occurs doing initialization that subsequent attempts to use the
+   * cluster result in an appropriate error.
+   *
+   * @jira_ticket JAVA-1220
+   * @test_category host:state
+   */
+  @Test(groups = "short")
+  public void should_detect_cluster_init_failure() {
```

Comment:
This passes even with 3.x in its current state, we should also have a more specific test that would have previously failed.

Reply:
This test will produce the failure that is detailed in JAVA-1221:

```java
  @Test(groups = "short")
  public void should_detect_cluster_init_failure_caused_by_bad_credentials() {
    Cluster cluster =
        Cluster.builder()
            .addContactPoints("127.0.0.1")
            .withCredentials("bad", "pass")
            .withNettyOptions(nonQuietClusterCloseOptions)
            .build();

    try {
      cluster.connect();
      fail("Should not have been able to connect.");
    } catch (RuntimeException e) {
      try {
        cluster.connect();
        fail("Should error when connect is called.");
      } catch (IllegalStateException e1) {
        assertThat(
            e1.getMessage().equals("Error during cluster initialization, please close and retry"));
      }
    } finally {
      cluster.close();
    }
  }
```

produces:

```
java.lang.IllegalArgumentException: initialArraySize cannot be negative but was: -2147483648
	at com.google.common.collect.CollectPreconditions.checkNonnegative(CollectPreconditions.java:39)
	at com.google.common.collect.Lists.newArrayListWithCapacity(Lists.java:186)
	at com.datastax.driver.core.HostConnectionPool.initAsync(HostConnectionPool.java:135)
	at com.datastax.driver.core.SessionManager.replacePool(SessionManager.java:362)
	at com.datastax.driver.core.SessionManager.maybeAddPool(SessionManager.java:397)
	at com.datastax.driver.core.SessionManager.createPools(SessionManager.java:129)
	at com.datastax.driver.core.SessionManager.initAsync(SessionManager.java:96)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:378)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
	at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
	at com.datastax.driver.core.ClusterInitTest.should_detect_cluster_init_failure(ClusterInitTest.java:303)
```

This would need to be updated to fire up a CCM cluster with authentication (see `AuthenticationTest` for reference).  To keep things simple you could just add this test to `AuthenticationTest`

Reply:
So this actually hits the exact same code path as my test does, just in a different way. I'm going to keep mine as it relies on simulacron rather than auth, and therefore will be shorter. The test above produces the same dump now that I've fixed it.

Reply:
I would still suggest adding my proposed test to `AuthenticationTest` since it produces the root issue of JAVA-1220.